### PR TITLE
ddt removal in ui hostgroup tests

### PR DESCRIPTION
**Pull request by sthirugn**
Test results:
```py
# nosetests tests/foreman/ui/test_hostgroup.py
......
----------------------------------------------------------------------
Ran 6 tests in 155.283s

OK
```

### CODE REVIEW

File: tests/foreman/ui/test_hostgroup.py [hunk 1]
```diff
@@ -57,12 +57,14 @@ def test_update_hostgroup(self, new_name):
...
             self.assertIsNotNone(self.hostgroup.search(name))
-            self.hostgroup.update(name, new_name=new_name)
-            self.assertIsNotNone(self.hostgroup.search(new_name))
+            for new_name in generate_strings_list(len1=4):
+                with self.subTest(new_name):
+                    self.hostgroup.update(name, new_name=new_name)
+                    self.assertIsNotNone(self.hostgroup.search(new_name))
+                    name = new_name  # for next iteration
 
-    @data(*generate_strings_list(len1=256))
```
Comment (sthirugn) [inline tests/foreman/ui/test_hostgroup.py]: this is a redundant test.  I replaced this with one test which uses `name = gen_string('alphanumeric', 256)`

Comment (oshtaier) [inline tests/foreman/ui/test_hostgroup.py]: I am little bit disagree with that and especially after your improvement that will make that particular test to be executed very fast (create only one entity and try to input every value).

I am absolutely fine for case when we have only three 'alpha', 'numeric' and 'alphanumeric to use 'alphanumeric', but not there

**Comment by oshtaier** [comment]
pending comment

File: tests/foreman/ui/test_hostgroup.py (hunk 1 above)
Comment (sthirugn) [inline tests/foreman/ui/test_hostgroup.py]: Yes, I understand that running all data points may not take so long to run.  But more valid question here is: Does it make sense to run this seven times?  The requirement here is that the name field should error out if the number of characters in the input is more than 255.  It doesn't matter if the input is utf8 or alphanumeric or html or anything.  Satellite considers all input as string and anything more than 255 characters in length will be errored out.  I am not sure what kind of bugs can be caught if we run this 7 times vs. 1 time?  Do you have an idea about that please?

Comment (oshtaier) [inline tests/foreman/ui/test_hostgroup.py]: if `Satellite considers all input as string` and does not distinguish utf8 and alphanumeric, so no questions here, but to honest we have such verifications in a huge amount of tests for some reason, right?

Comment (sthirugn) [inline tests/foreman/ui/test_hostgroup.py]: You are right, we are using it extensively using the helper invalid_values_list.  To maintain consistency, I am going to merge few tests here and use the invalid_values_list() here. do you agree?

Comment (oshtaier) [inline tests/foreman/ui/test_hostgroup.py]: perfect

Comment (sthirugn) [inline tests/foreman/ui/test_hostgroup.py]: ty @oshtaier will update this pr in a few

**Comment by sthirugn** [comment]
```py
# nosetests tests/foreman/ui/test_hostgroup.py
......
----------------------------------------------------------------------
Ran 5 tests in 169.021s

OK
```

**Comment by sthirugn** [comment]
ty @oshtaier all comments addressed

**Comment by oshtaier** [comment]
ACK

**Comment by abalakh** [comment]
ACK. Rebase, please

**Comment by sthirugn** [comment]
Rebased

**Comment by sthirugn** [comment]
part of https://github.com/SatelliteQE/robottelo/issues/2769

**Pull request merged**